OCPBUGS#12930: Created a register cluster section

dfitzmau opened the pull request:
[OCPBUGS-12930](https://issues.redhat.com/browse/OCPBUGS-12930)

Version(s):
4.14 through to 4.10

Link to docs preview:
* [Registering your disconnected cluster](https://63740--docspreview.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/opting-out-of-remote-health-reporting#insights-operator-register-unconnected-cluster_opting-out-remote-health-reporting)
* [AWS Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-restricted-networks-aws#installing-restricted-networks-aws-next-steps)
* [GCP Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned#next-steps_installing-restricted-networks-gcp-installer-provisioned)
* [Nutanix Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-restricted-networks-nutanix-installer-provisioned#next-steps)
* [Bare metal Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-restricted-networks-bare-metal#next-steps)
* [OpenStack Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-installer-restricted#next-steps)
* [vSphere Next Steps](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-restricted-networks-installer-provisioned-vsphere#next-steps_installing-restricted-networks-installer-provisioned-vsphere)

QE review:
- [x] QE has approved this change.
* Reached out to Joao Bastos Fula for an additional QE review.

Additional information:
* https://console.redhat.com/openshift/register
* [Opting out of remote health reporting](https://docs.openshift.com/container-platform/4.12/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html#insights-operator-new-pull-secret_opting-out-remote-health-reporting)
* [How does the subscriptions service show my subscription data?](https://access.redhat.com/documentation/en-us/subscription_central/2023/html/getting_started_with_the_subscriptions_service/con-how-does-subscriptionwatch-show-data_assembly-viewing-understanding-subscriptionwatch-data-ctxt)

dfitzmau commented:
SME Query. I see a restricted doc set for [z/VM on IBM Z® and IBM® LinuxONE](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z). I do not see an `[opt out of remote health reporting](https://63740--docspreview.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/opting-out-of-remote-health-reporting#opting-out-remote-health-reporting_opting-out-remote-health-reporting) link in the Next Steps section, so I'm wondering if the IBM docs need a link to *Registering an unconnected cluster*?

dfitzmau commented on support/remote_health_monitoring/opting-out-of-remote-health-reporting.adoc:
```diff
@@ -30,6 +30,13 @@ include::modules/telemetry-consequences-of-disabling-telemetry.adoc[leveloffset=
+[role="_additional-resources"]
+.Additional resources
+* xref:../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.adoc#telemetry-consequences-of-disabling-telemetry_opting-out-remote-health-reporting[Consequences of disabling remote health reporting]
+* link:https://access.redhat.com/documentation/en-us/subscription_central/2023/html/getting_started_with_the_subscriptions_service/con-how-does-subscriptionwatch-show-data_assembly-viewing-understanding-subscriptionwatch-data-ctxt[How does the subscriptions service show my subscription data?](Getting Started with the Subscription Service)
```
SME query: I included this link so a customer has more info about subscription services. The information pertaining to this doc set is broad, so is the link needed?

sferich888 replied:
Having the link is good

sferich888 commented:
> SME Query. I see a restricted doc set for [z/VM on IBM Z® and IBM® LinuxONE](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z). I do not see an `[opt out of remote health reporting](https://63740--docspreview.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/opting-out-of-remote-health-reporting#opting-out-remote-health-reporting_opting-out-remote-health-reporting) link in the Next Steps section, so I'm wondering if the IBM docs need a link to *Registering an unconnected cluster*?

Yes those platforms also need links to those sections, as they can often be disconnected.

sferich888 reviewed (changes requested):
I don't see anything in this PR that provides instruction to register clusters with automatic reporting disabled (or not possible do to the disconnect nature of the cluster). 

We need to instruct users to Regis and report clusters if they are disconnected.

dfitzmau commented:
> > SME Query. I see a restricted doc set for [z/VM on IBM Z® and IBM® LinuxONE](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z). I do not see an `[opt out of remote health reporting](https://63740--docspreview.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/opting-out-of-remote-health-reporting#opting-out-remote-health-reporting_opting-out-remote-health-reporting) link in the Next Steps section, so I'm wondering if the IBM docs need a link to _Registering an unconnected cluster_?
> 
> Yes those platforms also need links to those sections, as they can often be disconnected.

Thanks, @sferich888 . I added the links to about 4 IBM doc sets. Registering an unconnected cluster should be preceded with "If necessary" or "optional", just like the links for "opt out of remote health reporting"? Example: [Next steps IBM Power Virtual Server](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_powervs/installing-restricted-networks-ibm-power-vs#next-steps_installing-ibm-power-vs-restricted).

dfitzmau commented:
> > > SME Query. I see a restricted doc set for [z/VM on IBM Z® and IBM® LinuxONE](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z). I do not see an `[opt out of remote health reporting](https://63740--docspreview.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/opting-out-of-remote-health-reporting#opting-out-remote-health-reporting_opting-out-remote-health-reporting) link in the Next Steps section, so I'm wondering if the IBM docs need a link to _Registering an unconnected cluster_?
> > 
> > 
> > Yes those platforms also need links to those sections, as they can often be disconnected.
> 
> Thanks, @sferich888 . I added the links to about 4 IBM doc sets. Registering an unconnected cluster should be preceded with "If necessary" or "optional", just like the links for "opt out of remote health reporting"? Example: [Next steps IBM Power Virtual Server](https://63740--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_powervs/installing-restricted-networks-ibm-power-vs#next-steps_installing-ibm-power-vs-restricted).

Additionally, the *Next Steps* section for some IBM doc set were missing the "opt out of remote health reporting" links, so I added these. Maybe they were missed?

dfitzmau commented:
> automatic reporting disabled

Thanks, @sferich888 

> I don't see anything in this PR that provides instruction to register clusters with automatic reporting disabled (or not possible do to the disconnect nature of the cluster).
> 
> We need to instruct users to Regis and report clusters if they are disconnected.

Hi @openshift-ci-robot . Thanks for this info. I cannot find any steps necessary for this `automatic usage reporting`. Would you be OK to provide them?

sferich888 commented:
> > automatic reporting disabled
> 
> Thanks, @sferich888
> 
> > I don't see anything in this PR that provides instruction to register clusters with automatic reporting disabled (or not possible do to the disconnect nature of the cluster).
> > We need to instruct users to Regis and report clusters if they are disconnected.
> 
> Hi @openshift-ci-robot . Thanks for this info. I cannot find any steps necessary for registering clusters with `automatic usage reporting` disabled. Would you be OK to provide them?

You wouldn't because that isn't possible. The very nature of automatic reporting requires a connection. 
Thus registration (reporting) is done manually by users using https://console.redhat.com/openshift/register. 

tl;dr you create a cluster (and if its network isolated/disconnected or you disable reporting/telemetry) then you *must* manually report (register) your cluster using https://console.redhat.com/openshift/register 

cc: @tkatarki what further guidance can the BU provide here?  Without this in our docs; a lot of the business data we look at about clusters is incomplete.

dfitzmau commented:
> > > automatic reporting disabled
> > 
> > 
> > Thanks, @sferich888
> > > I don't see anything in this PR that provides instruction to register clusters with automatic reporting disabled (or not possible do to the disconnect nature of the cluster).
> > > We need to instruct users to Regis and report clusters if they are disconnected.
> > 
> > 
> > Hi @openshift-ci-robot . Thanks for this info. I cannot find any steps necessary for registering clusters with `automatic usage reporting` disabled. Would you be OK to provide them?
> 
> You wouldn't because that isn't possible. The very nature of automatic reporting requires a connection. Thus registration (reporting) is done manually by users using https://console.redhat.com/openshift/register.
> 
> tl;dr you create a cluster (and if its network isolated/disconnected or you disable reporting/telemetry) then you _must_ manually report (register) your cluster using https://console.redhat.com/openshift/register
> 
> cc: @tkatarki what further guidance can the BU provide here? Without this in our docs; a lot of the business data we look at about clusters is incomplete.

Thanks, Eric.

I found the following doc set, so I'm wondering if content in this doc set relates to our content?

[Converting a connected cluster to a disconnected cluster](https://docs.openshift.com/container-platform/4.13/post_installation_configuration/connected-to-disconnected.html)

sferich888 commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,29 @@
+
+Register your unconnected {product-title} cluster on the {hybrid-console} so that your cluster is not impacted by the consequences listed in the section named "Consequences of disabling remote health reporting".
+
+By registering your unconnected cluster, the {product-title} Telemetry service can provide metrics about your cluster's health and update status.
```
IDK that this statement is accurate (if your unconnected) Telemtery can't report back to us (unless you provide it a way (with a proxy and some connection) to connect back to us.

sferich888 replied:
It might be better to say by registering your cluster your ensuring that proper entitlement usage reporting happens; and your not repeatedly asked by your account team for an audit... (while we can't say it this way; this is done for proper reporting and adherence to our subscription terms).

dfitzmau replied:
Updated with the following notice:

> By registering your unconnected cluster, you can continue to report your subscription usage to Red Hat. In turn, Red Hat can return accurate usage and capacity trends associated with your subscription, so that you can use the returned information to better organize subscription allocations across all of your resources.

sferich888 reviewed (approved):
/lgtm

dfitzmau commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,29 @@
+.Prerequisites
+
+* You are logged in to the {product-title} web console as `cluster-admin`.
+* You are logged in to the {hybrid-console}.
```
Hi @sferich888 . What if a customer is unable to access the console? Is it that they did not register a customer account or sign up to a specific subscription?

dfitzmau commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,29 @@
+.Procedure
+. From the navigation menu on the link:https://console.redhat.com/openshift[{hybrid-console}], select *Clusters*.
+
+. Select the *Register cluster* button.
```
Updated with a direct link but also added an optional step in the event they are on the home web page on the Hybrid Cloud Console.

jianlinliu commented:
/lgtm

JoaoFula commented:
I apologize on the delay on this, LGTM

nalhadef commented:
/label peer-review-in-progress
/remove-label peer-review-needed

nalhadef commented:
/lgtm, Darragh!

nalhadef commented:
/remove-label peer-review-in-progress
/label peer-review-done

mburke5678 commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,32 @@
+.Procedure
+. Go to the link:https://console.redhat.com/openshift/register[*Register disconnected cluster*] web page on the {hybrid-console}. 
+
+. Optional: To access the *Register disconnected cluster* web page from the home page of the {hybrid-console}, select the *Clusters* navigation menu item and then select the *Register cluster* button.
```
s/select/click

dfitzmau replied:
How about "got to" as per the IBM SG guidance:

![Screenshot from 2023-08-29 17-13-27](https://github.com/openshift/openshift-docs/assets/57954076/1d8663f2-a49c-4114-b3b9-0279db624dd7)

mburke5678 commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,32 @@
+
+. Enter your cluster's details in the provided fields on the *Register disconnected cluster* page. 
+
+. From the *Subscription settings* section of the page, select the radio buttons that apply to your Red Hat subscription offering. 
```
Per the ISG, avoid the use of _radio buttons_, refer to the control by the label. 
```suggestion
. From the *Subscription settings* section of the page, select the subscription settings that apply to your Red Hat subscription offering. 
```

mburke5678 commented on modules/insights-operator-register-unconnected-cluster.adoc:
```diff
@@ -0,0 +1,32 @@
+
+:_content-type: PROCEDURE
+[id="insights-operator-register-unconnected-cluster_{context}"]
+= Registering your unconnected cluster
```
s/unconnected/disconnected throughout ??

dfitzmau replied:
Well spotted. "Disconnected" is correct. I'll update all my instances.

mburke5678 commented on installing/installing_aws/installing-restricted-networks-aws.adoc:
```diff
@@ -201,4 +201,5 @@ include::modules/cluster-telemetry.adoc[leveloffset=+1]
 * Learn how to xref:../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[use Operator Lifecycle Manager (OLM) on restricted networks].
 * If the mirror registry that you used to install your cluster has a trusted CA, add it to the cluster by xref:../../openshift_images/image-configuration.adoc#images-configuration-cas_image-configuration[configuring additional trust stores].
 * If necessary, you can xref:../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.adoc#opting-out-remote-health-reporting_opting-out-remote-health-reporting[opt out of remote health reporting].
+* xref:../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.html#modules/insights-operator-register-unconnected-cluster_opting-out-remote-health-reporting[Registering your unconnected cluster]
```
Remove the `modules/` from each of the registering links. 
```suggestion
* xref:../../support/remote_health_monitoring/opting-out-of-remote-health-reporting.html#insights-operator-register-unconnected-cluster_opting-out-remote-health-reporting[Registering your unconnected cluster]
```

dfitzmau commented:
@mburke5678 . I incorporated all of your feedback. Would you be OK to continue your merge review?

dfitzmau commented:
@mburke5678 . Merge conflict corrected. Would you be OK to merge this PR after the build passes?

mburke5678 commented:
/cherrypick enterprise-4.10

mburke5678 commented:
/cherrypick enterprise-4.11

mburke5678 commented:
/cherrypick enterprise-4.13

mburke5678 commented:
/cherrypick enterprise-4.14
